substrate/frame/meta-tx/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Meta Tx
+//!
+//! TODO docs
+
+#![cfg_attr(not(feature = "std"), not_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+pub use pallet::*;
+
+use frame_support::{
+	dispatch::{DispatchInfo, GetDispatchInfo, PostDispatchInfo},
+	pallet_prelude::*,
+	traits::OriginTrait,
+};
+use frame_system::pallet_prelude::*;
+use sp_runtime::traits::TransactionExtension;
+use sp_runtime::traits::TransactionExtensionBase;
+use sp_runtime::traits::{Dispatchable, IdentifyAccount, Verify};
+
+/// Meta transaction type.
+// TODO: The `MetaTx` type is similar to `sp_runtime::generic::UncheckedExtrinsic`. However,
+// `MetaTx` cannot replace generic::UncheckedExtrinsic because we need to box the call type,
+// given that `MetaTx` is used as an argument type for a call.
```

Comment:
We could probably have this "destructured" in the extrinsic argument list if it looks funny because of the `UncheckedExtrinsic` similarity. I think it's fine.